qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvcc_factory.py

```diff
@@ -63,101 +99,110 @@ def __init__(
     def gradient(self, gradient: Optional[Union[GradientBase, Callable]]) -> None:
-        """Setter of the gradient function"""
-        self._gradient = gradient
+        """DEPRECATED. Use the constructor instead. Sets the gradient."""
+        self.minimum_eigensolver.gradient = gradient
 
-    @property
+    @property  # type: ignore
+    @deprecate_property(
+        "0.4", additional_msg="Use `minimum_eigensolver` and 'solver properties' instead."
+    )
     def expectation(self) -> Optional[ExpectationBase]:
-        """Getter of the expectation."""
-        return self._expectation
+        """DEPRECATED. Use ``minimum_eigensolver`` method and solver properties instead.
+        Returns expectation."""
+        return self.minimum_eigensolver.expectation
 
-    @expectation.setter
+    @expectation.setter  # type: ignore
+    @deprecate_property("0.4", additional_msg="Use the constructor instead.")
     def expectation(self, expectation: Optional[ExpectationBase]) -> None:
-        """Setter of the expectation."""
-        self._expectation = expectation
+        """DEPRECATED. Use the constructor instead. Sets expectation."""
+        self.minimum_eigensolver.expectation = expectation
 
-    @property
+    @property  # type: ignore
+    @deprecate_property(
+        "0.4", additional_msg="Use `minimum_eigensolver` and 'solver properties' instead."
+    )
     def include_custom(self) -> bool:
-        """Getter of the ``include_custom`` setting for the ``expectation`` setting."""
-        return self._include_custom
+        """DEPRECATED. Use ``minimum_eigensolver`` method and solver properties instead.
+        Getter of the ``include_custom`` setting for the ``expectation`` setting."""
+        return self.minimum_eigensolver.include_custom
 
-    @include_custom.setter
+    @include_custom.setter  # type: ignore
+    @deprecate_property("0.4", additional_msg="Use the constructor instead.")
     def include_custom(self, include_custom: bool) -> None:
-        """Setter of the ``include_custom`` setting for the ``expectation`` setting."""
-        self._include_custom = include_custom
+        """DEPRECATED. Use the constructor instead. Setter of the ``include_custom``
+        setting for the ``expectation`` setting."""
+        self.minimum_eigensolver.include_custom = include_custom
 
     @property
     def ansatz(self) -> Optional[UVCC]:
-        """Getter of the ansatz."""
+        """
+        Gets the ansatz of future VQEs produced by the factory.
+        If value is ``None`` it defaults to :class:`UVCCSD`.
+        """
         return self._ansatz
 
     @ansatz.setter
     def ansatz(self, ansatz: Optional[UVCC]) -> None:
-        """Setter of the ansatz. If ``None`` is passed, this factory will default to using the
-        :class:`~.UVCCSD` Ansatz."""
+        """
+        Sets the ansatz of future VQEs produced by the factory.
+        If set to ``None`` it defaults to :class:`UVCCSD`.
```

Comment:
Something like `Sets the ansatz to be used, by the factory, to create/set the solver. If None is provided then the factory internally will default to an instance of UVCCSD it creates itself.`